libs/knowledge-store/notebooks/support.ipynb

```diff
@@ -0,0 +1,596 @@
+    "    os.environ[\"ASTRA_DB_APPLICATION_TOKEN\"] = getpass.getpass(\"Enter Astra DB Application Token: \")\n",
+    "\n",
+    "    keyspace = input(\"Enter Astra DB Keyspace (Empty for default): \")\n",
+    "    if keyspace:\n",
+    "        os.environ[\"ASTRA_DB_KEYSPACE\"] = keyspace\n",
+    "    else:\n",
+    "        os.environ.pop(\"ASTRA_DB_KEYSPACE\", None)\n",
+    "else:\n",
+    "    print(\"Not in colab. Loading '.env' (see 'env.template' for example)\")\n",
+    "    import dotenv\n",
+    "    dotenv.load_dotenv()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initialize Cassio and Knowledge Store\n",
+    "With the environent variables set we initialize the Cassio library for talking to Cassandra / Astra DB.\n",
+    "We also create the `KnowledgeStore`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cassio\n",
+    "from langchain_openai import OpenAIEmbeddings\n",
+    "from ragstack_knowledge_store import CassandraKnowledgeStore\n",
+    "from ragstack_knowledge_store.directed_edge_extractor import DirectedEdgeExtractor\n",
+    "\n",
+    "cassio.init(auto=True)\n",
+    "embeddings = OpenAIEmbeddings()\n",
+    "SITE_PREFIX=\"astra_docs\"\n",
+    "knowledge_store = CassandraKnowledgeStore(\n",
+    "    embeddings,\n",
+    "    edge_extractors = [\n",
+    "        DirectedEdgeExtractor.for_hrefs_to_urls(),\n",
+    "    ],\n",
+    "    node_table=f\"{SITE_PREFIX}_nodes\",\n",
+    "    edge_table=f\"{SITE_PREFIX}_edges\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<cassandra.cluster.ResultSet at 0x17fb2da90>"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from cassio.config import check_resolve_session, check_resolve_keyspace\n",
+    "session = check_resolve_session()\n",
+    "keyspace = check_resolve_keyspace()\n",
+    "session.execute(f\"DROP TABLE IF EXISTS {keyspace}.{SITE_PREFIX}_nodes\")\n",
+    "session.execute(f\"DROP TABLE IF EXISTS {keyspace}.{SITE_PREFIX}_edges\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load the Documents\n",
+    "Finally, we fetch pages and write them to the knowledge store in batches of 50."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Fetching pages:  22%|##1       | 300/1368 [00:51<06:37,  2.68it/s]Error fetching https://docs.datastax.com/en/cql/dse/reference/cql-commands/alter-materialized-view.html with attempt 1/3: Cannot connect to host docs.datastax.com:443 ssl:default [nodename nor servname provided, or not known]. Retrying...\n",
```

Comment:
Can we have a run without errors ?

Reply:
Maybe? I don't want to keep rerunning this since it is a lot of embeddings / writes to Astra. I'd like to get this in as a baseline and then work on a final version once I add MMR, etc.